### PR TITLE
GH-3623: Integration flow refactoring

### PR DESCRIPTION
**Pull request by oxcafedead**
For GH-3623.

Please review if this looks sane.

I decided to make it the simplest, just moved all factory methods to the main interface (thanks to private methods in interfaces no additional refactoring required).

I could have easily missed something very stupid and obvious, so apologizing in advance.

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/dsl/IntegrationFlows.java [hunk 1]
```diff
@@ -51,7 +51,10 @@
  * @since 5.0
  *
  * @see org.springframework.integration.dsl.context.IntegrationFlowBeanPostProcessor
+ *
+ * @deprecated Since version 6.0, this factory is not recommended to use as it shall be removed in the following versions. Use fluent API methods straight from {@link IntegrationFlow} interface.
  */
+@Deprecated(since = "6.0", forRemoval = true)
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/dsl/IntegrationFlows.java:57]: You know when I deprecate something, I prefer to delegate to the new API, so when we change something in that API, the deprecated one is going to be affected automatically.

**Review by artembilan — CHANGES_REQUESTED** [review]
I find this as an impressive work.
I''m not sure what else you could miss with such an overhaul everywhere!

Please, address those my two comments.
I also would like to see some `What's New` bullet to say that this was changed.
See `whats-new.adoc`.

Thank you!

**Comment by oxcafedead** [comment]
@artembilan makes sense, sure! I have updated the PR

**Review by artembilan — CHANGES_REQUESTED** [review]

File: src/reference/asciidoc/whats-new.adoc [hunk 2]
```diff
@@ -73,4 +73,8 @@ The `DefaultLockRepository` can now be supplied with a `PlatformTransactionManag
 
 === TCP/IP Changes
 
-The `lookupHost` property of the `AbstractConnectionFactory` and `DatagramPacketMessageMapper` is now set to `false` by default to avoid delays in the environments where DNS is not configured.
\ No newline at end of file
+The `lookupHost` property of the `AbstractConnectionFactory` and `DatagramPacketMessageMapper` is now set to `false` by default to avoid delays in the environments where DNS is not configured.
+
+=== IntegrationFlow API changes
+
+The `IntegrationFlows` factory is now marked as deprecated in favor of the fluent API available in the `IntegrationFlow` interface itself. The factory class will be removed in the future releases.
```
[the remark this answers is not in the document]
Comment (oxcafedead) [reply src/reference/asciidoc/whats-new.adoc]: Sure, I have updated the docs, but I am not sure if we need this "See ... for more information." section to be consistent 😄

[the remark this answers is not in the document]
Comment (artembilan) [reply src/reference/asciidoc/whats-new.adoc]: No problem! I’ll review it later on: I see we have a couple missed for other changes…

**Comment by artembilan** [comment]
I've just merged this as is.

Will clean up anything if meet eventually.

@oxcafedead ,

thank you very much for the contribution; looking forward for more!

The Migration Guide note on the matter: https://github.com/spring-projects/spring-integration/wiki/Spring-Integration-5.x-to-6.0-Migration-Guide#deprecated-the-integrationflows-class

**Pull request merged**